internal/drivers/manager.go

```diff
@@ -306,7 +306,7 @@ func (m *Manager) StartInstancePurger(ctx context.Context, maxAgeBusy, maxAgeFre
 
 // Provision returns an instance for a job execution and tags it as in use.
 // This method and BuildPool method contain logic for maintaining pool size.
-func (m *Manager) Provision(ctx context.Context, poolName, runnerName, serverName, ownerID, resourceClass string, env *config.EnvConfig, query *types.QueryParams) (*types.Instance, error) {
+func (m *Manager) Provision(ctx context.Context, poolName, runnerName, serverName, ownerID, resourceClass string, env *config.EnvConfig, query *types.QueryParams, gitspaceAgentConfig *types.GitspaceAgentConfig) (*types.Instance, error) {
```

Comment:
This function is tied with pooling logic today (not ideal but it's been there since the beginning). In case of GCP/AWS, we keep a pool of VMs ready. This method selects a VM from that pool and adds another in the pool to recuperate that.
In hosted for nomad specifically we don't have a pool but we keep pools for GCP/AWS in some cases.
Let's say if we add pooling back for nomad, this function would behave in a weird way since it's checking for the pool but adding a non-gitspace VM in the pool.

Let's dissociate gitspaces from pooling completely so that each time a request comes in, we create a new VM. We also need to make sure gitspaces only uses the nomad pool since the implementation is only present there. For gitspaces, let's just call the setup directly in a separate function so we can keep it very simple and free from errors.

Something like this in the beginning of the function:
```
m.runnerName = runnerName
m.liteEnginePath = env.LiteEngine.Path
m.tmate = types.Tmate(env.Tmate)
pool := m.poolMap[poolName]
if pool == nil {
  return nil, fmt.Errorf("provision: pool name %q not found", poolName)
}
if pool.Driver != "nomad" { // double check the name
  return nil, fmt.Errorf("incorrect pool, gitspaces is only supported on nomad")
}
if gitspaceAgentConfig != nil {
   inst, err = m.setupInstance(ctx, pool, serverName, ownerID, resourceClass, true, gitspaceAgentConfig)
   return inst, err
}
```